skyportal/handlers/api/followup_request.py

```diff
@@ -458,7 +470,19 @@ def put(self, request_id):
             payload={"obj_id": followup_request.obj_id},
         )
 
-        followup_request.instrument.api_class.update(followup_request)
+        response = followup_request.instrument.api_class.update(followup_request)
+
+        transaction = FacilityTransaction(
+            request=http.serialize_requests_request(response.request),
+            response=http.serialize_requests_response(response),
+            followup_request=followup_request,
+            initiator=self.associated_user_object,
+        )
+
+        DBSession().add(followup_request)
+        DBSession().add(transaction)
+        DBSession().commit()
```

Comment:
With two `commit()` calls, this can still fail after the first commit, leading to a partially-updated DB state. Perhaps remove the first commit in this handler or replace it with a flush call if necessary

Reply:
This is intended. We want to differentiate between whether the request failed before it was submitted to the observatory or after.

Reply:
fixed